### PR TITLE
Clean up thinp tests

### PR DESCRIPTION
**Pull request by nirs**
Clean up thinp tests after moving all the code from vm.py.

Changes:
- unify test modules names (thinp_test.py , thinp_monitor_test.py)
- Fix and improve some tests
- Update stale TODOs
- Remove unneeded tests that already tested elswhere
- Add some public API tests

Before:
```
$ wc -l tests/virt/{drive_extension,thinp}_test.py 
  766 tests/virt/drive_extension_test.py
  620 tests/virt/thinp_test.py
 1386 total
```

After:
```
$ wc -l tests/virt/thinp*_test.py 
  772 tests/virt/thinp_monitor_test.py
  212 tests/virt/thinp_test.py
  984 total
```

402 lines deleted, tests coverage unchanged (76%).

### CODE REVIEW

**Comment by nirs** [comment]
This tests only change, tested on my fork.

File: tests/virt/thinp_monitor_test.py [hunk 1]
```diff
@@ -339,11 +336,16 @@ def test_extend_no_allocation(tmp_config):
                 diskReplicate={
                     'format': 'cow',
                     'diskType': DISK_TYPE.BLOCK,
+                    'size': 2 * GiB,
                 },
             ),
-            block_info(allocation=2 * GiB, physical=2 * GiB),
+            # Libvirt reports same allocation and phisical for files, so we
```
Comment (vjuranek) [inline tests/virt/thinp_monitor_test.py:339]: "phisical" -> "physical"

File: tests/virt/thinp_monitor_test.py [hunk 2]
```diff
@@ -19,7 +19,9 @@
 #
 
 """
-Testng extension during replication may not be implemented yet.
+TODO:
+
+Testng replication flows:
```
Comment (vjuranek) [inline tests/virt/thinp_monitor_test.py:24]: "Testng: -> "Testing"

File: tests/virt/thinp_monitor_test.py [hunk 3]
```diff
@@ -19,7 +19,9 @@
 #
```
Comment (vjuranek) [inline tests/virt/thinp_monitor_test.py:19]: "Turns out" -> "It turns out"
"seems that" -> "It seems that"

File: tests/virt/thinp_test.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 from vdsm.virt import thinp
```
Comment (vjuranek) [inline tests/virt/thinp_test.py:24]: "What we did test is _monitored_volumes()" probably better to use "In the past we tested _monitored_volumes()"

**Review by vjuranek — COMMENTED** [review]
Few typos, code LGTM.

**Comment by barpavel** [comment]
Looks good, there is a small typo in PR description
> the all the code

all the code

**Comment by nirs** [comment]
@vjuranek all the typos fixed and I applied your suggested text changes. No code change, ready for merge when tests all checks complete.

**Review by barpavel — APPROVED** [review]

**Review by vjuranek — APPROVED** [review]

**Pull request merged**